FWCore/Catalog/src/FileLocator.cc

```diff
@@ -33,201 +31,42 @@ namespace {
-        } else if (argTokens[0] == "destination") {
-          m_destination = argTokens[1];
-        }
-      }
-    }
-
-    if (m_protocols.empty()) {
-      throw cms::Exception("TrivialFileCatalog",
-                           "TrivialFileCatalog::connect: protocol was not supplied in the contact string");
-    }
-
-    std::ifstream configFile;
-    configFile.open(m_filename.c_str());
-
-    if (!configFile.good() || !configFile.is_open()) {
-      throw cms::Exception("TrivialFileCatalog",
-                           "TrivialFileCatalog::connect: Unable to open trivial file catalog " + m_filename);
-    }
-
-    configFile.close();
-
-    tinyxml2::XMLDocument doc;
-    auto loadErr = doc.LoadFile(m_filename.c_str());
-    if (loadErr != tinyxml2::XML_SUCCESS) {
-      throw cms::Exception("TrivialFileCatalog")
-          << "tinyxml file load failed with error : " << doc.ErrorStr() << std::endl;
-    }
-    /* trivialFileCatalog matches the following xml schema
-	 FIXME: write a proper DTD
-	 <storage-mapping>
-	 <lfn-to-pfn protocol="direct" destination-match=".*"
-	 path-match="lfn/guid match regular expression"
-	 result="/castor/cern.ch/cms/$1"/>
-	 <pfn-to-lfn protocol="srm"
-	 path-match="lfn/guid match regular expression"
-	 result="$1"/>
-	 </storage-mapping>
-    */
-    auto rootElement = doc.RootElement();
-    /*first of all do the lfn-to-pfn bit*/
-    for (auto el = rootElement->FirstChildElement("lfn-to-pfn"); el != nullptr;
-         el = el->NextSiblingElement("lfn-to-pfn")) {
-      parseRuleTrivialCatalog(el, m_directRules_trivialCatalog);
-    }
-
-    /*Then we handle the pfn-to-lfn bit*/
-    for (auto el = rootElement->FirstChildElement("pfn-to-lfn"); el != nullptr;
-         el = el->NextSiblingElement("pfn-to-lfn")) {
-      parseRuleTrivialCatalog(el, m_inverseRules);
+  std::string FileLocator::pfn(std::string const& ilfn) const {
+    //check if ilfn is an authentic LFN
+    if (ilfn.compare(0, 7, kLFNPrefix) != 0) {
+      return "";
     }
+    return applyRules(m_directRules, m_protocol, ilfn);
   }
 
-  void FileLocator::init(edm::CatalogAttributes const& input_dataCatalog,
+  void FileLocator::init(CatalogAttributes const& catalogAttributes,
                          unsigned iCatalog,
                          std::string const& storageDescriptionPath) {
     Service<SiteLocalConfig> localconfservice;
-    edm::CatalogAttributes aCatalog = input_dataCatalog;
-    if (input_dataCatalog.empty()) {
-      if (!localconfservice.isAvailable()) {
-        cms::Exception ex("FileCatalog");
-        ex << "edm::SiteLocalConfigService is not available";
-        ex.addContext("Calling edm::FileLocator::init()");
-        throw ex;
-      }
+    if (!localconfservice.isAvailable()) {
+      cms::Exception ex("FileCatalog");
+      ex << "edm::SiteLocalConfigService is not available";
+      ex.addContext("Calling edm::FileLocator::init()");
+      throw ex;
+    }
```

Comment:
The earlier code seems to allow `SiteLocalConfig` service to be unavailable if necessary information is specified explicitly. I'd suggest to preserve that behavior.